core/controllers/admin.py

```diff
@@ -220,78 +259,146 @@ def get(self):
         })
 
     @acl_decorators.can_access_admin_page
-    def post(self):
+    def post(self) -> None:
         """Handles POST requests."""
-        action = self.normalized_payload.get('action')
+        assert self.user_id is not None
+        # Here we use cast because we are narrowing down the type of
+        # 'normalized_payload' from Dict[str, Any] to a particular
+        # TypedDict that was defined according to the schemas. So that
+        # the type of fetched values is not considered as Any type.
+        payload_data = cast(
+            AdminHandlerNormalizePayloadDict, self.normalized_payload
+        )
+        action = payload_data['action']
         try:
             result = {}
-            if action == 'reload_exploration':
-                exploration_id = self.normalized_payload.get('exploration_id')
+            if action and action == 'reload_exploration':
```

Comment:
why is this needed?

Wouldn't we get the same result for this condition even if the action is `None`?

Reply:
Fixed!